cpp/src/io/parquet/page_data.cu

```diff
@@ -1838,7 +1860,19 @@ __global__ void __launch_bounds__(block_size) gpuDecodePageData(
   bool const has_repetition = s->col.max_level[level_type::REPETITION] > 0;
 
   // if we have no work to do (eg, in a skip_rows/num_rows case) in this page.
-  if (s->num_rows == 0 && !(has_repetition && is_bounds_page(s, min_row, num_rows))) { return; }
+  //
+  // corner case: in the case of lists, we can have pages that contain "0" rows if the current row
+  // starts before this page and ends after this page:
+  //       P0        P1        P2
+  //  |---------|---------|----------|
+  //        ^------------------^
+  //      row start           row end
+  // P1 will contain "0" rows
```

Comment:
```suggestion
  // P1 will contain 0 rows
```

Reply:
The idea with the quotes was to imply zero-but-not-really.

Reply:
I'm not sure if I understand the "but not really" portion. That seems important to understand the scope of the bug. Can you elaborate? I may just be missing some knowledge about the Parquet spec.

Reply:
My understanding is that no rows start (or end) in such page. So it does not (fully) contain any rows, but it does contain data that we need to parse (and apparently we didn't, before this PR).
That's at least how I understand the ASCII art in the comment above :)

Reply:
Got it. That helps. Thanks! I’ll approve.

Reply:
We don’t need to remove the air-quotes, but maybe the comment can be updated with the description @vuule wrote above.